diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/app/bin/smalldlg.c ken-source-new/app/bin/smalldlg.c
--- ken-source-orig/app/bin/smalldlg.c	2026-08-27 13:18:24.000000000 -0400
+++ ken-source-new/app/bin/smalldlg.c	2026-08-27 15:59:37.016520667 -0400
@@ -218,7 +218,7 @@
 
 static paramTextData_t aboutTextData = { 50, 10 };
 
-#define DESCRIPTION N_("XTrackCAD is a CAD (computer-aided design) program for designing model railroad layouts.")
+#define DESCRIPTION N_("Debug XtrackCad build from KensTest 7040:6f9ba15ab942.\n\nXTrackCAD is a CAD (computer-aided design) program for designing model railroad layouts.")
 static paramData_t aboutPLs[] = {
 #define I_ABOUTDRAW				(0)
 	{   PD_BITMAP, NULL, "about", PDO_NOPSHUPD, NULL, NULL, 0 },
