diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/app/bin/smalldlg.c ken-source-new/app/bin/smalldlg.c
--- ken-source-orig/app/bin/smalldlg.c	2026-09-17 08:00:00.000000000 -0400
+++ ken-source-new/app/bin/smalldlg.c	2026-09-17 10:00:26.976823751 -0400
@@ -224,7 +224,7 @@
 
 static paramTextData_t aboutTextData = { 50, 10 };
 
-#define DESCRIPTION N_("XTrackCAD is a CAD (computer-aided design) program for designing model railroad layouts.")
+#define DESCRIPTION N_("Flatpak build from GTK3V2MAIN 7324:027d8429d784\n\nXTrackCAD is a CAD (computer-aided design) program for designing model railroad layouts.")
 static paramData_t aboutPLs[] = {
 #define I_ABOUTDRAW				(0)
 	{   PD_BITMAP, NULL, "about", PDO_NOPSHUPD, NULL, NULL, 0 },
