diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/app/bin/custom.c ken-source-new/app/bin/custom.c
--- ken-source-orig/app/bin/custom.c	2026-08-10 05:08:30.000000000 -0400
+++ ken-source-new/app/bin/custom.c	2026-08-19 10:37:24.052182094 -0400
@@ -42,10 +42,12 @@
  * \var sProdNameLower
  * Lowercase product name
  *
- * For flatpak builds, this should not contain any periods.
+ * For local flatpak builds, this should not contain any periods.
+ *
+ * For flathub flatpak builds, this should only be "xtrkcad".
  */
 #if XTRKCAD_BUILD_FLATPAK
-char * sProdNameLower = product "-" XTRKCAD_VERSION_ID;
+char * sProdNameLower = product XTRKCAD_VERSION_ID;
 #else
 char * sProdNameLower = product "-" XTRKCAD_VERSION;
 #endif
diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/app/lib/CMakeLists.txt ken-source-new/app/lib/CMakeLists.txt
--- ken-source-orig/app/lib/CMakeLists.txt	2026-08-10 05:08:30.000000000 -0400
+++ ken-source-new/app/lib/CMakeLists.txt	2026-08-20 05:03:01.303626909 -0400
@@ -78,8 +78,8 @@
 			)
 		install(FILES
 			xtrkcad.xml
-			DESTINATION "share/applications"
-			RENAME ${XTRKCAD_BIN}.xml
+			DESTINATION "share/mime/packages"
+			RENAME io.sourceforge.xtrkcad_fork.${XTRKCAD_BIN}.xml
 			)
 		install(FILES
 			xtrkcad.metainfo.xml
diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/app/lib/xtrkcad.desktop.in ken-source-new/app/lib/xtrkcad.desktop.in
--- ken-source-orig/app/lib/xtrkcad.desktop.in	2026-08-06 09:36:32.000000000 -0400
+++ ken-source-new/app/lib/xtrkcad.desktop.in	2026-08-24 04:21:59.231123423 -0400
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Name=XTrackCAD @XTRKCAD_VERSION@
 Comment=Design model railroad layouts
-Exec=@XTRKCAD_BIN@
+Exec=@XTRKCAD_BIN@ %f
+MimeType=application/x-xtrkcad;
 Icon=@XTRKCAD_BIN@
 Terminal=false
 Type=Application
-Categories=Graphics
+Categories=Graphics;
diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/app/lib/xtrkcad.flatpak.desktop ken-source-new/app/lib/xtrkcad.flatpak.desktop
--- ken-source-orig/app/lib/xtrkcad.flatpak.desktop	2026-08-17 16:38:43.000000000 -0400
+++ ken-source-new/app/lib/xtrkcad.flatpak.desktop	2026-08-24 04:23:32.851046874 -0400
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Name=XTrackCAD
 Comment=Design model railroad layouts
-Exec=run-xtrkcad.sh
-Icon=io.sourceforge.xtrkcad_fork.xtrkcad-5_4_0
+Exec=run-xtrkcad.sh %f
+MimeType=application/x-xtrkcad;
+Icon=io.sourceforge.xtrkcad_fork.xtrkcad
 Terminal=false
 Type=Application
-Categories=Graphics
+Categories=Graphics;
diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/app/lib/xtrkcad.metainfo.xml ken-source-new/app/lib/xtrkcad.metainfo.xml
--- ken-source-orig/app/lib/xtrkcad.metainfo.xml	2026-08-17 16:38:43.000000000 -0400
+++ ken-source-new/app/lib/xtrkcad.metainfo.xml	2026-08-20 18:26:55.109711715 -0400
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop-application">
-  <id>io.sourceforge.xtrkcad_fork.xtrkcad-5_4_0</id>
-  <name>XTrackCAD-5_4_0</name>
+  <id>io.sourceforge.xtrkcad_fork.xtrkcad</id>
+  <name>XTrackCAD</name>
   <summary>Design model railroad layouts</summary>
   <description>
     <p>XTrackCAD is a CAD (computer-aided design) program for designing model railroad layouts.</p>
@@ -11,7 +11,7 @@
     <release version="5.4.0" date="2026-08-11"/>
   </releases>
   <content_rating type="oars-1.1" />
-  <launchable type="desktop-id">io.sourceforge.xtrkcad_fork.xtrkcad-5_4_0.desktop</launchable>
+  <launchable type="desktop-id">io.sourceforge.xtrkcad_fork.xtrkcad.desktop</launchable>
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>GPL-3.0</project_license>
   <url type="bugtracker">https://sourceforge.net/p/xtrkcad-fork/bugs/</url>
@@ -26,7 +26,7 @@
       <caption>Example port_dinllean.xtc</caption>
     </screenshot>
   </screenshots>
-  <developer id="io.sourceforge.xtrkcad_fork.xtrkcad-5_4_0">
+  <developer id="io.sourceforge.xtrkcad_fork.xtrkcad">
     <name>xtrkcad developers</name>
   </developer>
 </component>
diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/app/wlib/gtk3lib/unix/ixpaths.c ken-source-new/app/wlib/gtk3lib/unix/ixpaths.c
--- ken-source-orig/app/wlib/gtk3lib/unix/ixpaths.c	2026-08-11 08:30:22.000000000 -0400
+++ ken-source-new/app/wlib/gtk3lib/unix/ixpaths.c	2026-08-19 10:29:23.376579909 -0400
@@ -143,7 +143,7 @@
 	}
 	strcpy( ep, "LIB" );
 	ep = getenv( envvar );
-	//printf("DEBUG: %s=%s\n", envvar, ep);
+	printf("DEBUG: %s=%s\n", envvar, ep);
 	searchDirs[ option++ ] = g_strdup( ep );
 
 	// option 2: relative path to share subdir
diff -uNrb -X /home/kshaffer/.diff_ignore --no-dereference ken-source-orig/CMakeLists.txt ken-source-new/CMakeLists.txt
--- ken-source-orig/CMakeLists.txt	2026-08-11 08:30:22.000000000 -0400
+++ ken-source-new/CMakeLists.txt	2026-08-19 08:40:16.392230318 -0400
@@ -143,9 +143,11 @@
 #
 # Note: flatpak builds cannot have periods in the name since periods delimit
 #       the namespace.
-if (XTRKCAD_BUILD_FLATPAK)
+if (XTRKCAD_BUILD_FLATPAK AND Flatpak_FOUND)
 	string(REPLACE "." "_" XTRKCAD_VERSION_ID "${XTRKCAD_VERSION}")
 	set(XTRKCAD_BIN "xtrkcad-${XTRKCAD_VERSION_ID}")
+elseif(XTRKCAD_BUILD_FLATPAK)
+	set(XTRKCAD_BIN "xtrkcad")
 else()
 	set(XTRKCAD_BIN "xtrkcad-${XTRKCAD_VERSION}")
 endif()
